File: xamarin/Samples/Showcase-Android/Activities/WebContentListActivity.cs
using System.Collections;
using Android.App;
using Android.OS;
using Android.Views;
using Android.Widget;
using Com.Liferay.Mobile.Screens.Base.List;
using Com.Liferay.Mobile.Screens.Webcontent.List;

namespace ShowcaseAndroid
{
    [Activity]
    public class WebContentListActivity : Activity, IBaseListListener
    {
        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);
            SetContentView(Resource.Layout.WebContentListView);

            WebContentListScreenlet webContentListScreelet = 
                (WebContentListScreenlet) FindViewById(Resource.Id.web_content_list_screenlet);
            webContentListScreelet.Listener = this;
        }

        /* IBaseListListener */

        public void Error(Java.Lang.Exception p0, string p1)
		{
			System.Diagnostics.Debug.WriteLine($"WebContent list error: {p0}");
		}

		public void OnListItemSelected(Java.Lang.Object p0, View p1)
		{
			Toast.MakeText(this, $"Item selected {p0}", ToastLength.Short).Show();
		}

		public void OnListPageFailed(int p0, Java.Lang.Exception p1)
		{
			System.Diagnostics.Debug.WriteLine($"WebContent list page failed: {p0}");
		}

		public void OnListPageReceived(int p0, int p1, IList p2, int p3)
		{
            Toast.MakeText(this, $"WebContent list page recived: {p3} entries", ToastLength.Short).Show();
		}

	}
}

File: android/samples/test-app/src/main/java/com/liferay/mobile/screens/testapp/WebContentListActivity.java
package com.liferay.mobile.screens.testapp;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import com.liferay.mobile.screens.base.list.BaseListListener;
import com.liferay.mobile.screens.viewsets.defaultviews.DefaultAnimation;
import com.liferay.mobile.screens.webcontent.WebContent;
import com.liferay.mobile.screens.webcontent.list.WebContentListScreenlet;
import java.util.List;

/**
 * @author Javier Gamarra
 */
public class WebContentListActivity extends ThemeActivity implements BaseListListener<WebContent> {

    private WebContentListScreenlet webContentListScreenlet;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.web_content_display_list);

        webContentListScreenlet = findViewById(R.id.web_content_list_screenlet);
        webContentListScreenlet.setListener(this);
    }

    @Override
    public void onListPageFailed(int startRow, Exception e) {
        error(getString(R.string.page_error), e);
    }

    @Override
    protected void onResume() {
        super.onResume();

        webContentListScreenlet.loadPage(0);
    }

    @Override
    public void onListPageReceived(int startRow, int endRow, List<WebContent> entries, int rowCount) {
        info(rowCount + " " + getString(R.string.rows_received_info) + " " + startRow);
    }

    @Override
    public void onListItemSelected(WebContent webContent, View view) {

        Intent intent = getIntentWithTheme(WebContentDisplayActivity.class);
        intent.putExtra("articleId", webContent.getArticleId());
        DefaultAnimation.startActivityWithAnimation(this, intent);
    }

    @Override
    public void error(Exception e, String userAction) {

    }
}
